libevm/lib/service_evm.go

```diff
@@ -129,17 +141,18 @@ func (s *Service) EvmApply(params EvmParams) (error, *EvmResult) {
 		}
 		blockContext = params.getBlockContext()
 		chainConfig  = defaultChainConfig()
+		tracer       = logger.NewStructLogger(nil)
```

Comment:
The trace parameters need to be passed to the tracer here to have an effect. I recommend adding a function to the `TraceParams` struct to produce a geth-compatible type, similar to what `getBlockContext` does and then do:
`tracer = logger.NewStructLogger(params.TxTraceParams.getTraceConfig())` here. Something like:
```
func (t *TraceParams) getTraceConfig() logger.Config {
	return logger.Config{
		EnableMemory:     t.EnableMemory,
		DisableStack:     ...,
		DisableStorage:   ...,
		EnableReturnData: ...,
	}
}
```